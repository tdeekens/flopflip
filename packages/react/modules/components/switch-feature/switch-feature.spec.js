import React from 'react';
import { shallow } from 'enzyme';
import ToggleFeature from '../toggle-feature';
import SwitchFeature from './switch-feature';

const FeatureComponent = props => (
  <React.Fragment>FeatureComponent {props.nbr}</React.Fragment>
);
FeatureComponent.propTypes = { nbr: PropTypes.number.isRequired };
FeatureComponent.displayName = 'FeatureComponent';

describe('rendering', () => {
  describe('with a feature match', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <SwitchFeature>
          <ToggleFeature isFeatureEnabled variate="foo">
            <FeatureComponent nbr={1} />
          </ToggleFeature>
          <ToggleFeature isFeatureEnabled={false} variate="bar">
            <FeatureComponent nbr={2} />
          </ToggleFeature>
        </SwitchFeature>
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the `FeatureComponent`', () => {
      expect(wrapper).toRender(FeatureComponent);
    });

    it('should supply `variate` to the `ToggleFeature`', () => {
      expect(wrapper.find(ToggleFeature)).toHaveProp('variate', 'foo');
    });
  });

  describe('without a feature match', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <SwitchFeature>
          <ToggleFeature isFeatureEnabled={false} variate="foo">
            <FeatureComponent nbr={1} />
          </ToggleFeature>
          <ToggleFeature isFeatureEnabled={false} variate="bar">
            <FeatureComponent nbr={2} />
          </ToggleFeature>
        </SwitchFeature>
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not render any `FeatureComponent`', () => {
      expect(wrapper).not.toRender(FeatureComponent);
    });
  });

  describe('with multiple feature matchs', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <SwitchFeature>
          <ToggleFeature isFeatureEnabled variate="foo">
            <FeatureComponent nbr={1} />
          </ToggleFeature>
          <ToggleFeature isFeatureEnabled variate="bar">
            <FeatureComponent nbr={2} />
          </ToggleFeature>
        </SwitchFeature>
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should not render one `FeatureComponent`', () => {
      expect(wrapper).toRenderElementTimes(FeatureComponent, 1);
    });

    it('should not render the first `FeatureComponent`', () => {
      expect(wrapper.find(ToggleFeature)).toHaveProp('variate', 'foo');
    });
  });
});
