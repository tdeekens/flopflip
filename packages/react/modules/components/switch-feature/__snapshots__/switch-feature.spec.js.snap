// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with a feature match should match snapshot 1`] = `
<ToggleFeature
  isFeatureEnabled={true}
  render={null}
  toggledComponent={null}
  untoggledComponent={null}
  variate="foo"
>
  <FeatureComponent
    nbr={1}
  />
</ToggleFeature>
`;

exports[`rendering with multiple feature matchs should match snapshot 1`] = `
<ToggleFeature
  isFeatureEnabled={true}
  render={null}
  toggledComponent={null}
  untoggledComponent={null}
  variate="foo"
>
  <FeatureComponent
    nbr={1}
  />
</ToggleFeature>
`;

exports[`rendering without a feature match should match snapshot 1`] = `""`;
